ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -946,19 +949,18 @@ - (void)showViewController:(ORKStepViewController *)viewController goForward:(BO
     
     ORKWeakTypeOf(self) weakSelf = self;
     [self.pageViewController setViewControllers:@[viewController] direction:direction animated:animated completion:^(BOOL finished) {
-        ORKStrongTypeOf(weakSelf) strongSelf = weakSelf;
```

Comment:
Did you test leaving this in and instead adding a check with an early exit if `weakSelf == nil`. By strongifying self, the code shouldn't dealloc *during* this block.